### PR TITLE
Swapping cuda 11 with cuda 12 in packing and publishing pipeline

### PR DESCRIPTION
**Pull request by jchen351**
### Description
Since we are switch to cuda 12 as the default on official releases, we would need to swap all current packaging pipeline and publishing pipeline to set the cuda 12 as the default.

### Motivation and Context
This is needed for current 1.19 release.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/nuget-cuda-publishing-pipeline.yml [hunk 1]
```diff
@@ -17,9 +19,9 @@ parameters:
 variables:
   - name: ArtifactFeed
     ${{ if eq(parameters.isReleaseBuild, false) }}:
-      value: ORT-Nightly
+      value: rt-cuda-11-nightly
```
Comment (tianleiwu) [inline tools/ci_build/github/azure-pipelines/nuget-cuda-publishing-pipeline.yml:22]: ort-cuda-11-nightly

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 2]
```diff
@@ -114,15 +113,15 @@ stages:
...
     SpecificArtifact: ${{ parameters.SpecificArtifact }}
     BuildId: ${{ parameters.BuildId }}
 
 - template: stages/nuget-combine-cuda-stage.yml
   parameters:
       DoCompliance: ${{ parameters.DoCompliance }}
-      CudaVersion: 11.8
-      docker_base_image: 'nvidia/cuda:11.8.0-cudnn8-devel-ubi8'
+      CudaVersion: 12.2
+      docker_base_image: 'nvidia/cuda:12.2.2-cudnn8-devel-ubi8'
```
Comment (tianleiwu) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:124]: Please use cudnn 9 with cuda 12.

Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:124]: Cudnn9 needs cuda 12.3.2 there are many other place need to be changes as well.

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml [hunk 3]
```diff
@@ -80,12 +80,11 @@ variables:
 - name: ReleaseVersionSuffix
   value: ''
 - name: win_trt_version
-  value: 11.8
```
Comment (jywu-msft) [inline tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:83]: should not be changed. i was told that this yml is for producing 11.8 packages. there is another one for 12.x

**Review by jywu-msft — CHANGES_REQUESTED** [review]
don't merge this until we understand the intended setup.

File: tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml (hunk 3 above)
Comment (jchen351) [reply tools/ci_build/github/azure-pipelines/c-api-noopenmp-packaging-pipelines.yml:83]: This one is for default cuda package version, which is now cuda 12.2

**Comment by jywu-msft** [comment]
closing this PR. it's no longer needed. 1.19.1 was released with cuda 12.x as default for gpu packages

**Comment by tianleiwu** [comment]
I think we revisit this. See https://github.com/microsoft/onnxruntime/issues/21849.

**Pull request closed without merging**